webdriver/tests/execute_script/promise.py

```diff
@@ -0,0 +1,80 @@
+import pytest
+
+from tests.support.asserts import assert_dialog_handled, assert_error, assert_success
+
+
+def execute_script(session, script, args=None):
+    if args is None:
+        args = []
+    body = {"script": script, "args": args}
+
+    return session.transport.send(
+        "POST", "/session/{session_id}/execute/sync".format(
+            session_id=session.session_id),
+        body)
+
+
+def test_promise_resolve(session):
+    response = execute_script(session, """
+        return Promise.resolve('foobar');
+        """)
+    assert_success(response, "foobar")
+
+
+def test_promise_resolve_delayed(session):
+    response = execute_script(session, """
+        return new Promise(
+            (resolve) => setTimeout(
+                () => resolve('foobar'),
+                50
+            )
+        );
+        """)
+    assert_success(response, "foobar") 
+
+
+def test_promise_reject(session):
+    response = execute_script(session, """
+        return Promise.reject(new Error('my error'));
+        """)
+    assert_error(response, "unknown error")
+    assert response.body["value"]["message"] == "Error: my error"
```

Comment:
The error message is entirely implementation-defined, so we can’t test this.

Reply:
removed